leaderelection/leaderelection.go

```diff
@@ -91,25 +88,19 @@ func (m *Manager) IsLeader() bool {
 	return m.leader.IsLeader()
 }
 
-// Run leader election is blocking until a leader is being elected or timeout has reached.
-func (m *Manager) Run(ctx context.Context) error {
+// Run leader election is blocking until a FirstLeaderDeadline timeout has reached.
+func (m *Manager) Run(ctx context.Context) {
 	timeoutCtx, cancel := context.WithTimeout(ctx, FirstLeaderDeadline)
 	defer cancel()
 
 	go m.leader.Run(ctx)
 	m.log.Infof("started leader election")
 
-	select {
-	case <-m.block:
-		m.log.Infof("new leader has been elected")
-	case <-timeoutCtx.Done():
-		m.log.Warnf("timeout - no leader has been elected for %s", FirstLeaderDeadline.String())
-	}
-
-	return nil
+	<-timeoutCtx.Done()
```

Comment:
You can use `time.After` as the context has no true usage.

Reply:
Following question:
Does the call to `go m.leader.Run(ctx)` move the `ctx` context to "done" when it finishes?
If so, we can maintain the select syntax we had before, and add the `time.After` as another case.

Reply:
It will not change the status of our ctx, as a new ctx is created in the leaderelection package, and canceling it will only be propagated down the context hierarchy.

Reply:
So let's make sure that's indeed the case, and if so, there is no advantage of using `time.After`, instead we can just use `time.Sleep`